Integrations/EDL/EDL.py

```diff
@@ -0,0 +1,371 @@
+    """
+    iocs: List[dict] = []
+    if not last_found_len:
+        last_found_len = total_fetched
+    while last_found_len == PAGE_SIZE and limit and total_fetched < limit:
+        fetched_iocs = demisto.findIndicators(query=indicator_query, page=next_page, size=PAGE_SIZE).get('iocs')
+        iocs.extend(fetched_iocs)
+        last_found_len = len(fetched_iocs)
+        total_fetched += last_found_len
+        next_page += 1
+    return iocs, next_page
+
+
+# def find_and_consolidate_ips_to_limit(indicator_query, iocs, limit):
+#     """
+#     Groups IPs and find new ones afterward in a loop until limit is reached
+#     """
+#     pre_consolidate_iocs_len = len(iocs)
+#     iocs, next_page = find_new_ips_and_consolidate(indicator_query, iocs, limit)
+#     while len(iocs) != pre_consolidate_iocs_len:
+#         pre_consolidate_iocs_len = len(iocs)
+#         iocs, next_page = find_new_ips_and_consolidate(indicator_query, iocs, limit, next_page)
+#     return iocs
+#
+#
+# def find_new_ips_and_consolidate(indicator_query, iocs, limit, next_page=0):
+#     """
+#     Finds new IPs and consolidates them
+#     """
+#     last_iocs_found, next_page = find_indicators_to_limit_loop(indicator_query, limit, total_fetched=len(iocs),
+#                                                                next_page=next_page)
+#     iocs = consolidate_ips(iocs)
+#     return iocs, next_page
+#
+#
+# def consolidate_ips(iocs):
+#     """
+#     Groups together ips in a list of strings
+#     """
+#     try:
+#         iocs = sorted(iocs, key=lambda ip: struct.unpack('!L', inet_aton(ip))[0])
+#     except OSError:
+#         demisto.debug('Failed to consolidate IPs')
+#         return iocs
+
+
+def create_csv_out_list(cache_dict: dict) -> list:
+    """
+    Creates a csv output result
+    """
+    csv_headers = cache_dict.pop(CSV_FIRST_LINE_KEY, '')
+    values_list = list(cache_dict.values())
+    if csv_headers:
+        values_list.insert(0, csv_headers)
+
+    return values_list
+
+
+def create_values_out_dict(iocs: list, out_format: str) -> dict:
+    """
+    Create a dictionary for output values using the selected format
+    """
+    out_format_func = {
+        FORMAT_TEXT: format_ioc_text_out,
+        FORMAT_JSON_SEQ: format_ioc_json_seq_out,
+        FORMAT_CSV: format_ioc_csv_out
+    }
+    return create_formatted_values_out_dict(iocs, out_format, out_format_func.get(out_format, str))
+
+
+def create_formatted_values_out_dict(iocs: list, out_format: str, out_format_func: Callable) -> dict:
+    """
+    Create a dictionary for output values formatted in the selected out_format
+    """
+    out_dict = {}
+    if out_format == FORMAT_JSON:
+        iocs_list = [ioc for ioc in iocs]
+        return {'iocs_list': json.dumps(iocs_list, indent=4)}
+    else:
+        for ioc in iocs:
+            value = ioc.get('value')
+            if value:
+                out_dict[value] = out_format_func(ioc)
```

Comment:
write a function that will replace out_format_func and create_values_out_dict,
it will get the args `ioc` and `out_format` and it will do the translation.